src/isomorphic/classic/element/ReactElementValidator.js

```diff
@@ -143,7 +121,15 @@ function validateChildKeys(node, parentType) {
     for (var i = 0; i < node.length; i++) {
       var child = node[i];
       if (ReactElement.isValidElement(child)) {
-        validateExplicitKey(child, parentType);
+        const getMainExplicitKeyErrorMessage = getExplicitKeyErrorMessage.bind(
+          null,
+          parentType,
+        );
```

Comment:
I wonder if there's a way we could do this without having to create a new function for every child? What if `parentType` was passed to `validateExplicitKey` and it was called from there instead?

If not, we should move this outside of the for loop so the function isn't recreated on every iteration.